### PR TITLE
feat(ux): flatten wash reg push/pull into wash push/pull

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

Flatten `wash reg push` into `wash push`
Flatten `wash reg pull` into `wash pull`

See:
- https://github.com/wasmCloud/wash/issues/538

## Release Information

`next`

## Testing

Built on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### Acceptance or Integration

Altered the existing integration tests to use the new subcommands

### CODE REVIEW

File: src/common/registry_cmd.rs [hunk 1]
```diff
@@ -19,26 +19,50 @@ use wash_lib::registry::{
...
 
-const PROVIDER_ARCHIVE_FILE_EXTENSION: &str = ".par.gz";
-const WASM_FILE_EXTENSION: &str = ".wasm";
-
 pub(crate) const SHOWER_EMOJI: &str = "\u{1F6BF}";
+pub(crate) const PROVIDER_ARCHIVE_FILE_EXTENSION: &str = ".par.gz";
+pub(crate) const WASM_FILE_EXTENSION: &str = ".wasm";
+
+#[derive(Parser, Debug, Clone)]
+pub(crate) struct AuthOpts {
```
Comment (connorsmith256) [inline src/common/registry_cmd.rs]: Are these worth moving to wash-lib and making public?

Comment (vados-cosmonic) [reply src/common/registry_cmd.rs]: Ah yeah I can definitely do that, I think it makes sense, maybe rename this to `RegistryAuthOpts`

- [x] move (& rename) `AuthOpts` to `wash-lib`

**Comment by vados-cosmonic** [comment]
Weird issue with `clap` -- getting messages saying the crate is undeclared after moving the `Registry...Command`s into `wash-lib`.

[EDIT] Looks like this has to do with making `clap` optional -- should be able to change the test script

**Review by connorsmith256 — APPROVED** [review]
LGTM!

**Pull request merged**